### PR TITLE
introduce nats-streaming

### PR DESCRIPTION
**Pull request by andriisoldatenko**
**Which issue this PR fixes**:

fix https://github.com/astronomer/issues/issues/1415

**Summary of changes**:

add new image nats-streaing

**Which images are updated by this PR?**:

- [x] nats-streaming

**Checklist** (required)

Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.

- [x] version.txt is updated in the changed image(s)

If a security scan fails for an unchanged image...

- [ ]  a fix has been applied OR...
- [ ]  an [issue]() has been created

If adding a new image:

- [x] the directory has the same name you intend it to be published as, less a preceding "ap-"
- [x] the directory includes a file "version.txt", with version matching the underlying software version
- [x] the directory includes a file "cve-whitelist.yaml", which may be empty
- [x] the file .github/PULL_REQUEST_TEMPLATE.md is updated to include the image in the checklist
- [c] execute the script .circleci/generate_circleci_config.py, commit changes to .circleci/config.yml

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -116,6 +116,28 @@ workflows:
             branches:
               only: master
 
+      - build:
```
Comment (sjmiller609) [inline .circleci/config.yml:119]: Please make this change in config.yml.j2 then run the script in that directory.

Comment (andriisoldatenko) [reply .circleci/config.yml:119]: i did this way, i've added only 1 new dir, so no need to change config.yaml.j2
this loop take care on it dynamically: https://github.com/astronomer/ap-vendor/blob/b5bc6155f13d79d810a4c4ccc19a7d11c76d21b2/.circleci/config.yml.j2#L6-L5

Comment (sjmiller609) [reply .circleci/config.yml:119]: ah right! thank you

**Comment by sjmiller609** [comment]
@schnie you will have to do the needful wrt to setting up new image repo on dockerhub and attaching permissions like you normally do for new vendored images

**Comment by schnie** [comment]
@sjmiller609 @andriisoldatenko I created the dockerhub repo and our bot should have access to write to it.

**Comment by andriisoldatenko** [comment]
restarted job flow.

**Comment by andriisoldatenko** [comment]
@schnie 
```
+ curl --silent -f -lSL https://index.docker.io/v1/repositories/astronomerinc/ap-nats-streaming/tags/latest
curl: (22) The requested URL returned error: 404 NOT FOUND
+ tag_and_push latest
+ docker tag ap-nats-streaming astronomerinc/ap-nats-streaming:latest
+ docker push astronomerinc/ap-nats-streaming:latest
The push refers to a repository [docker.io/astronomerinc/ap-nats-streaming]
fac92e89: Preparing 
c65d87f2: Preparing 
denied: requested access to the resource is denied
Exited with code exit status 1
CircleCI received exit code 1
```

**Comment by schnie** [comment]
:facepalm: I forgot the `ap-` prefix. Should be there now!

**Comment by andriisoldatenko** [comment]
@schnie @sjmiller609 it works! well done!
```
docker pull astronomerinc/ap-nats-streaming:0.18.0
0.18.0: Pulling from astronomerinc/ap-nats-streaming
df20fa9351a1: Already exists
638813c8986e: Already exists
cba66f363cd2: Already exists
Digest: sha256:107977935387bad7f6aed445ac4b2e6e036803c3850cd9f649f3ff2876f8ce5d
Status: Downloaded newer image for astronomerinc/ap-nats-streaming:0.18.0
docker.io/astronomerinc/ap-nats-streaming:0.18.0
```

**Pull request merged**